### PR TITLE
Updated robot simulator to package.xml format 2.  Added test depends to workaround test failure

### PR DESCRIPTION
**Pull request by shaun-edwards**
Workaround for issue catkin/catkin_tools#285.  Should allow travis test jobs to succeed (or else not fail due to roslaunch test issues).

Note: This also required updating the package.xml format to version 2.

### CODE REVIEW

**Comment by shaun-edwards** [comment]
Tests now PASS!

@gavanderhoorn, can you please review updates to package.xml format 2?

**Comment by gavanderhoorn** [comment]
Hm, I'm not too much of a fan of changing things just to accommodate a (temporary) bug in a build tool, but if this is needed, then so be it.

It would seem I'm partially responsible for us using `catkin_tools` instead of `catkin_make` anyway (ros-industrial/industrial_ci#3), so ..

File: industrial_robot_simulator/package.xml [hunk 1]
```diff
@@ -9,18 +9,17 @@
...
-  <run_depend>industrial_robot_client</run_depend>
-  <run_depend>industrial_msgs</run_depend>
-  <run_depend>python-rospkg</run_depend>
-  <run_depend>rospy</run_depend>
+
+  <depend>std_msgs</depend>
+  <depend>sensor_msgs</depend>
+  <depend>control_msgs</depend>
+  <depend>trajectory_msgs</depend>
+  <depend>industrial_msgs</depend>
```
Comment (gavanderhoorn) [inline industrial_robot_simulator/package.xml:14]: It is recommended to sort all dependencies alphabetically.

File: industrial_robot_simulator/package.xml [hunk 2]
```diff
@@ -9,18 +9,17 @@
...
+
+  <exec_depend>industrial_robot_client</exec_depend>
+  <exec_depend>python-rospkg</exec_depend>
+  <exec_depend>rospy</exec_depend>
```
Comment (gavanderhoorn) [inline industrial_robot_simulator/package.xml:21]: same here (and below).

**Comment by gavanderhoorn** [comment]
+1 on the changes (ie: they are syntactically correct), but -1 for why we are doing this. As @ipa-mdl already commented in ros-industrial/ros_canopen#155, this can / should be reverted once `catkin_tools` 0.4.0 is released. I'd rather we switch to `catkin_make` for the time being (don't know how feasible that is though, @130s?).

**Comment by mathias-luedtke** [comment]
> Note: This also required updating the package.xml format to version 2.

Actually this is an improvement.

It should work with format 1 as well, if you add `industrial_robot_client` as a `build_depend`.
(It feels odd, but it enforces the correct order)

File: industrial_robot_simulator/package.xml [hunk 3]
```diff
@@ -9,18 +9,17 @@
...
+
+  <test_depend version_gte="1.9.55">roslaunch</test_depend>
+  <test_depend>industrial_robot_client</test_depend>
```
Comment (mathias-luedtke) [inline industrial_robot_simulator/package.xml:29]: This line should be a separate commit. This simplifies the revert.

**Comment by shaun-edwards** [comment]
I split the commits (one for upgrade to format 2, one for the workaround).  I sorted the depends alphabetically.  

I'm not excited about committing a work-around, so I added a comment about the reason for the `test_depends`.

@gavanderhoorn, ok to merge?

**Comment by gavanderhoorn** [comment]
@shaun-edwards: looks like Travis test failed, despite the work-around?

**Comment by shaun-edwards** [comment]
Looks like it may be related to the number of parallel jobs, see [here](https://docs.travis-ci.com/user/customizing-the-build/#Limiting-Concurrent-Builds).  I've set the number of parallel jobs to 1.  This should solve the problem of multiple tests using the same socket port.

**Comment by shaun-edwards** [comment]
@gavanderhoorn, all fixed now...ok to merge?

**Comment by gavanderhoorn** [comment]
+1 then.

But lets log an issue to revert 40fa0da once `catkin_tools` has been fixed.
